sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -811,7 +809,11 @@ public void deleteDataset(String projectId, String datasetId)
                                   "BigQuery insertAll error, retrying: %s",
                                   ApiErrorExtractor.INSTANCE.getErrorMessage(e)));
                           try {
-                            sleeper.sleep(backoff1.nextBackOffMillis());
+                            long nextBackOffMillis = backoff1.nextBackOffMillis();
+                            if (nextBackOffMillis == BackOff.STOP) {
+                              throw e;
```

Comment:
Can we add a unit test for this behavior ?

Reply:
Done.